src/coreclr/vm/threads.h

```diff
@@ -3776,6 +3777,11 @@ class Thread
     // See ThreadStore::TriggerGCForDeadThreadsIfNecessary()
     bool m_fHasDeadThreadBeenConsideredForGCTrigger;
 
+    CLRRandom m_random;
```

Comment:
This is reverting a change that was made here: https://github.com/dotnet/runtime/pull/99989/files#r1531477052

I suspect it will be worthwhile to have a thread-local random number generator to avoid lock contention in the Alloc() code path, but perhaps this should now be a pointer that is allocated on demand rather than eagerly allocated in the Thread object.
@jkotas - any thoughts since it looks like you were advocating for the original removal of this field?

Reply:
@jkoritzinsky - your thoughts as well?

Reply:
I think it should be a free-standing static. There is no reason for all random statics to be centralized on the Thread type.

> this should now be a pointer that is allocated on demand rather than eagerly allocated in the Thread object.

Yes. Or switch to a more compact random implementation. 220 bytes of state for pseudo random number generator sounds like an overkill. E.g. [Xorshiro](https://github.com/dotnet/runtime/blob/ce1477b5df9b378f1dfcbb9194aeed3932ecfc03/src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs#L31) used by BCL currently has 16 bytes of state.